### PR TITLE
Add Supermarket program

### PR DESCRIPTION
**Pull request by sarafuentes3025**

### CODE REVIEW

File: src/test/java/com/sfuentes/SupermarketTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package com.sfuentes;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class SupermarketTest {
```
Comment (aandraca) [inline src/test/java/com/sfuentes/SupermarketTest.java:10]: I have more test cases in mind, for example:
What happens if I add the same item name twice? will that give me the correct amount?
What happens if I add a null key? or a null price?
etc.
Try to come up with all the possible test cases.

**Review by aandraca — CHANGES_REQUESTED** [review]

File: src/main/java/com/sfuentes/Supermarket.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  Map<String, Double> items;
+
+  public Supermarket(Map<String, Double> items) {
+    this.items = items;
+  }
+
+  public String getTotalPurchases() {
+    double totalToPay = 0;
+    for (double item : items.values())
```
[the remark this answers is not in the document]
Comment (sarafuentes3025) [reply src/main/java/com/sfuentes/Supermarket.java]: Fixed.

File: src/main/java/com/sfuentes/Supermarket.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+public class Supermarket {
+
+  Map<String, Double> items;
+
+  public Supermarket(Map<String, Double> items) {
+    this.items = items;
+  }
+
+  public String getTotalPurchases() {
```
[the remark this answers is not in the document]
Comment (sarafuentes3025) [reply src/main/java/com/sfuentes/Supermarket.java]: Fixed.

File: src/main/java/com/sfuentes/Supermarket.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    double totalToPay = 0;
+    for (double item : items.values())
+      totalToPay += item;
+
+    return "Number of items: " + items.size() + ", Total to pay: " + totalToPay;
```
[the remark this answers is not in the document]
Comment (sarafuentes3025) [reply src/main/java/com/sfuentes/Supermarket.java]: Fixed.

File: src/test/java/com/sfuentes/SupermarketTest.java (hunk 1 above)
Comment (sarafuentes3025) [reply src/test/java/com/sfuentes/SupermarketTest.java:7]: fixed

File: src/test/java/com/sfuentes/SupermarketTest.java [hunk 5]
```diff
@@ -2,37 +2,42 @@
...
-    items.put("cookies: ", 5d);
-    items.put("beans: ", 4d);
-    items.put("milk: ", 2d);
-    items.put("granola: ", 6d);
-    items.put("tea: ", 13d);
-    items.put("deodorant: ", 12d);
-
-    Supermarket test = new Supermarket(items);
-    assertEquals("Number of items: 11, Total to pay: 75.7", test.getTotalPurchases());
+  void calculateTotal_duplicatedItems_integerResult() {
```
Comment (aandraca) [inline src/test/java/com/sfuentes/SupermarketTest.java:10]: ...integerResult doesn't provide enough information on what is the expected behavior.
something more like sumAllValues

**Review by aandraca — CHANGES_REQUESTED** [review]

File: src/test/java/com/sfuentes/SupermarketTest.java (hunk 5 above)
Comment (sarafuentes3025) [reply src/test/java/com/sfuentes/SupermarketTest.java]: fix

**Review by aandraca — APPROVED** [review]

**Pull request merged**